src/components/tag/tag.jsx

```diff
@@ -27,15 +30,21 @@ export default function Tag(props) {
         keyEvent.preventDefault();
         // falls through
       case 'ArrowRight':
-        setInputHandled(false);
-        setTabActiveStates(props.readOnly ? [true] : [false, true]);
+        if (!props.readOnly && tabActiveStates[0] === true && tabActiveStates[1] === false) {
+          keyEvent.stopPropagation();
+          setInputHandled(false);
+          setTabActiveStates([false, true]);
+        }
         break;
       case 'ArrowUp':
         keyEvent.preventDefault();
```

Comment:
We shouldn't always prevent the default. We should only prevent the default when the Tag component is consuming the event. Otherwise, we can have a Tag in the middle of a page and the user wants to scroll right but needs to tab out of the Tag. That seems unintuitive. I wonder if there is a rule of thumb for this kind of thing. My intuition is pointing to: "if we use it then we prevent default, if we don't use it then we do the default (because we don't care about the event)"